### PR TITLE
WIP - Tracking of equivalent functions

### PR DESCRIPTION
**Pull request by justin-lovell**
Initial commit to track equivalence and scope of DCS API's

At some point, I would like to refer to the completeness of the gRPC API surface area. This would allow the scope to be worked on incrementally.

Feel free to suggest updates to the paragraph of not committing to a roadmap. There are a lot of functions which should not be implemented.

### CODE REVIEW

File: STATUS.md [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+- [ ] `getStaticObjects`
+- [x] `getAirbases`
+- [ ] `getPlayers`
+- [ ] `getServiceProviders`
+- [ ] `addRefPoint`
+- [ ] `getRefPoints`
+- [x] `getMainRefPoint` ("Bullseye")
+- [ ] `getCountryCoalition`
+
+### Coord Singleton
```
Comment (rurounijones) [inline STATUS.md:53]: I don't see a need for these in the public API. The cients should only be sending LL and we do the conversions internally.

File: STATUS.md [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+- [ ] `LLtoLO`
+- [ ] `LOtoLL`
+- [ ] `LLtoMGRS`
+- [ ] `MGRStoLL`
+
+### Env Singleton
```
Comment (rurounijones) [inline STATUS.md:59]: `info`, `warning`, `error` likewise seem like things clients should not be sending (although I could see an argument for putting everything into one logfile. I would still not be particulart in favour of it).

File: STATUS.md [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+- [ ] `get_stat`
+- [x] <strike>`get_name`</strike> (will not be implemented -- returned part of `get_player_info`)
+- [x] <strike>`get_slot`</strike> (will not be implemented -- returned part of `get_player_info`)
+- [ ] `set_slot`
+- [x] `force_player_slot`
+- [ ] `lua2json`
+- [ ] `json2lua`
+- [ ] `dostring_in`
+- [ ] `log`
+- [ ] `trace`
```
Comment (rurounijones) [inline STATUS.md:108]: Similary to above, I don't think these should be parts of the public APIs

File: STATUS.md [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+#### GameGUI API
+  - [ ] `load_mission`
+  - [ ] `load_next_mission`
+
+
+### Timer Service
+- [x] `getTime`
+- [x] `getAbsTime`
+- [x] `getTime0`
+- [ ] `scheduleFunction` (TODO -- should this be implemented?)
```
Comment (rurounijones) [inline STATUS.md:119]: I would be against it. Clients can all the APIs when they need to,

**Comment by justin-lovell** [comment]
All good @rurounijones - first in the collection phase, marked what is implemented, and going to eliminate the never.

File: STATUS.md [hunk 5]
```diff
@@ -0,0 +1,487 @@
...
+### Mission Commands Singleton
+- [ ] `addCommand`
+- [ ] `addSubMenu`
+- [ ] `removeItem`
+- [ ] `addCommandForCoalition`
+- [ ] `addSubMenuForCoalition`
+- [ ] `removeItemForCoalition`
+- [ ] `addCommandForGroup`
+- [ ] `addSubMenuForGroup`
+- [ ] `removeItemForGroup`
```
Comment (rurounijones) [inline STATUS.md:85]: Note: These have now been added but I didn't update this.

File: STATUS.md [hunk 6]
```diff
@@ -0,0 +1,487 @@
...
+### Coalitions Singleton
+- [ ] `addGroup`
+  - [ ] Sea
+    - [ ] Group
+    - [ ] Units
+    - [ ] Waypoints
+    - [ ] Tasks
+  - [ ] Ground
+    - [ ] Group
+    - [ ] Units
```
Comment (rurounijones) [inline STATUS.md:27]: Ground group and Units have been done (At least to the point that a NASAM site can be spawned)

**Comment by justin-lovell** [comment]
Closing as the branch/repository source has changed location ... which should allow distributed co-authoring - https://github.com/DCS-gRPC/rust-server/pull/125

**Pull request closed without merging**